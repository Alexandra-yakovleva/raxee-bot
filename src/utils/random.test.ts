import { getRandomItem, shuffleItems } from './random';

beforeEach(() => {
  jest.spyOn(global.Math, 'random').mockReturnValue(0.3);
});

afterEach(() => {
  jest.spyOn(global.Math, 'random').mockRestore();
});

describe('getRandomItem', () => {
  test('should return value', () => {
    expect(getRandomItem(['a', 'b', 'c', 'd'])).toBe('b');
  });

  test('should return firstItem when only one is presented', () => {
    expect(getRandomItem(['a'])).toBe('a');
  });
});

describe('shuffleItems', () => {
  test('should shuffle array', () => {
    expect(shuffleItems(['a', 'b', 'c', 'd'])).toMatchSnapshot();
  });

  test('should accept empty array', () => {
    expect(shuffleItems([])).toStrictEqual([]);
  });
});
