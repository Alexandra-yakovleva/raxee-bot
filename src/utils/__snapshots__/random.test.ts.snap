// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shuffleItems should shuffle array 1`] = `
Array [
  "d",
  "c",
  "a",
  "b",
]
`;
